File: YandexContest2021/src/hw4/Program4E.cpp


#include <fstream>
#include <map>
#include <set>
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    map <int, int> dict;

    ifstream input("C:\\Users\\1\\IdeaProjects\\Timus\\YandexContest2021\\src\\hw4\\Program4E_test.txt");


   int w;
   int h;
   int countButtons;
   input >> countButtons;
   for (int i = 0; i < countButtons;i++) {
       input >> w;
       input >> h;



        if (dict.find(w) == dict.end()) {//если нет такого ключа
            dict.insert(make_pair(w,h));
            //cout << 0 << " ";
        }
        else { //если такой ключ есть
            map <int, int> ::iterator it;
            it = dict.find(w);
            int oldHight = it->second;
            if (oldHight < h) {
                //Удаляем старый элемент
                dict.erase(w);
                dict.insert(make_pair(w, h));
            }

        }

    }

    long sum=0;

    int  hight;
    for (auto it = dict.cbegin(); it != dict.cend(); ++it) {

        hight = it->second;
        sum += hight;

    }
     cout << sum;


}

File: YandexContest2021/src/hw4/Program4E.java
package hw4;

import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Scanner;

public class Program4E {
    public static void main(String[] args) throws IOException {
        String fileName = "C:\\Users\\1\\IdeaProjects\\Timus\\YandexContest2021\\src\\hw4\\Program4E_test.txt";
        Path path = Paths.get(fileName);
        Scanner scanner = new Scanner(path);


        int countBricks= scanner.nextInt();
        HashMap<Integer,Integer> hashMap=new HashMap<>();

        for(int i=0;i<countBricks;i++){

            int w=scanner.nextInt();
            int h=scanner.nextInt();
            if(hashMap.containsKey(w)){
                if(hashMap.get(w)<h){
                    hashMap.put(w,h);
                }
            }else{
                hashMap.put(w,h);
            }

        }
        long sum=0;
        for(int e: hashMap.keySet()){
            sum+=hashMap.get(e);
        }
        System.out.println(sum);
    }
}
